### PR TITLE
Crash-Free Hours Cookbook

### PR DESCRIPTION
**Pull request by ejohnso49**
Doc describing Crash-Free Hours, how the metric relates to reliability, and how to start measuring it with Memfault.

## To Do
- [ ] Add examples of actual charts
- [ ] Add sections on Android and Linux support

### CODE REVIEW

File: docs/best_practices/metrics-for-fleet-reliability.mdx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+id: metrics-for-fleet-reliability
+title: Measuring Fleet Reliability With Crash-Free Hours
+sidebar_label: Measuring Fleet Reliability
+---
+
+Once devices leave the factory and are live in the field, visibility into their
+operation drastically decreases. Changes to the fleet like deploying a new
+update, releasing new hardware revisions, introducing new features all bring
```
Comment (noahp) [inline docs/best_practices/metrics-for-fleet-reliability.mdx:9]: harvard comma :sweat_smile: 

```suggestion
update, releasing new hardware revisions, introducing new features, all bring
```

**Review by noahp — APPROVED** [review]
:tada: Very nice! I'd suggest a quick re-read and see if you can tighten up a few of the longer/wordier paragraphs, most folks will be primarily interested in the "action" part of the article (but the background/context is very important)

File: docs/best_practices/metrics-for-fleet-reliability.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+are typically not dominated by random failures, but fixable ones as found in the
+left-hand section of the bathtub curve. A better metric would be one to
+calculate reliability across the range of the device's lifecycle
+
+The metric Memfault uses for this task is **crash-free hours**. Crash-free hours
+provides insight into device reliability across updates and measures the early
+phase failures. Before moving on to define crash-free hours, we need to define
+what marks an hour as crash-free.
+
+### Defining Unexpected Reboots
```
Comment (tyhoff) [inline docs/best_practices/metrics-for-fleet-reliability.mdx:88]: Here, you might want to mention that we didn't just create this up ourselves, but we are taking inspiration from "Crash-free users" and "Crash-free sessions" which are both widely used in the web and mobile space as measurable statistics. It's just that with IoT devices, our primarily interaction with them is time!

File: docs/best_practices/metrics-for-fleet-reliability.mdx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+- Device firmware updates
+- User initiated reboots
+- Fault handling
+- Watchdog resets
+
+The last two instances should be classified as unexpected reboots. A device
+should not encounter these during normal operation. The
+[reboot reason tracking subsystem](/docs/mcu/reboot-reason-tracking) is used
+exactly for this purpose. When tracking crash-free hours, it is very important
+to classify reboot reasons as accurate and precisely as possible.
```
[the remark this answers is not in the document]
Comment (ejohnso49) [reply docs/best_practices/metrics-for-fleet-reliability.mdx:103]: A user can define their own implementation of `memfault_reboot_reason_get` to customize this based on their hardware/setup. I'll link to the docs for more info on this

**Comment by tyhoff** [comment]
Making this "ready for review" since it's very much ready for review :D

**Comment by tyhoff** [comment]
I figured it was easier for me to push some updates, so I went ahead and did that, and I'm happy with where this is.

**Review by tyhoff — APPROVED** [review]

**Comment by victorbordo** [comment]
lgtm!

**Review by victorbordo — APPROVED** [review]

**Pull request merged**